neo4j-cli/aura/internal/subcommands/dataapi/graphql/corspolicy/allowedorigin/add.go

```diff
@@ -0,0 +1,97 @@
+package allowedorigin
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/neo4j/cli/common/clicfg"
+	"github.com/neo4j/cli/neo4j-cli/aura/internal/api"
+	"github.com/neo4j/cli/neo4j-cli/aura/internal/output"
+	"github.com/spf13/cobra"
+)
+
+func NewAddCmd(cfg *clicfg.Config) *cobra.Command {
+	const (
+		instanceIdFlag = "instance-id"
+		dataApiIdFlag  = "data-api-id"
+		awaitFlag      = "await"
+	)
+
+	var (
+		instanceId string
+		dataApiId  string
+		await      bool
+	)
+
+	cmd := &cobra.Command{
+		Use:   "add <origin>",
+		Short: "Adds a new allowed origin to the CORS policy",
+		Long: `This command adds a new allowed origin to the Cross-Origin Resource Sharing (CORS) policy of a GraphQL Data API.
+
+Updating the CORS policy of a GraphQL Data API is an asynchronous operation. Use the --await flag to wait for the GraphQL Data API to be ready. Once the status transitions from "updating" to "ready" you may begin to use your GraphQL Data API.
+
+Adding a new allowed origin to the CORS policy of a GraphQL Data API allows browsers to make requests to the GraphQL Data API from a web app that is served from the specified origin.`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			newOrigin := args[0]
+
+			existingOrigins, err := getGetExistingOrigins(cfg, dataApiId, instanceId)
+			if err != nil {
+				return err
+			}
+
+			for _, origin := range existingOrigins {
+				if origin == newOrigin {
+					cmd.Println("Origin already exists in allowed origins:", newOrigin)
```

Comment:
I'd consider returning an error instead. I _think_ it's more common in this case.